### PR TITLE
Add swap payment schedule adjustments

### PR DESCRIPTION
**Pull request by jodastephen**
Process the additional adjustment.
Clarify that dates are from the accrual schedule where unclear.

Reading closely, it seems that FpML will not use this, however it could produce different dates in certain edge cases, so it is a useful addition.

### CODE REVIEW

File: modules/basics/src/test/java/com/opengamma/strata/basics/schedule/SchedulePeriodTest.java [hunk 1]
```diff
@@ -41,7 +43,8 @@
   private static final LocalDate JUL_05 = date(2014, JULY, 5);
   private static final LocalDate JUL_17 = date(2014, JULY, 17);
   private static final LocalDate JUL_18 = date(2014, JULY, 18);
-  private static final LocalDate AUG_17 = date(2014, AUGUST, 17);
+  private static final LocalDate AUG_17 = date(2014, AUGUST, 17);  // Sunday
+  private static final LocalDate AUG_18 = date(2014, AUGUST, 18);  // Sunday
```
Comment (marc-opengamma) [inline modules/basics/src/test/java/com/opengamma/strata/basics/schedule/SchedulePeriodTest.java]: 18 Aug 2014 is Monday
